airflow/executors/kubernetes_executor.py

```diff
@@ -696,8 +701,7 @@ def try_adopt_task_instances(self, tis: List[TaskInstance]) -> List[TaskInstance
             for pod in pod_list.items:
                 self.adopt_launched_task(kube_client, pod, pod_ids)
         self._adopt_completed_pods(kube_client)
-        tis_to_flush.extend(pod_ids.values())
-        return tis_to_flush
+        return list(pod_ids.values())
```

Comment:
I took a look at the call sites of this, it seems that non of the actual usages strictly rely on this being a list; we should change the annotation to `Sequence[TaskInstance]` and return `pod_ids.values()` instead.

This would need to change a bunch more files so it’s OK to do this in a separate PR.

Reply:
I will look at that this week and create a new PR for it.